internal/aws/cwlogs/cwlog_client.go

```diff
@@ -42,23 +44,39 @@ const (
 type Client struct {
 	svc          cloudwatchlogsiface.CloudWatchLogsAPI
 	logRetention int64
+	tags         map[string]*string
+	accountID    string
+	region       string
 	logger       *zap.Logger
 }
 
 // Create a log client based on the actual cloudwatch logs client.
-func newCloudWatchLogClient(svc cloudwatchlogsiface.CloudWatchLogsAPI, logRetention int64, logger *zap.Logger) *Client {
+func newCloudWatchLogClient(svc cloudwatchlogsiface.CloudWatchLogsAPI, logRetention int64, tags map[string]*string, accountID string, region string, logger *zap.Logger) *Client {
 	logClient := &Client{svc: svc,
 		logRetention: logRetention,
+		tags:         tags,
+		accountID:    accountID,
+		region:       region,
 		logger:       logger}
 	return logClient
 }
 
 // NewClient create Client
-func NewClient(logger *zap.Logger, awsConfig *aws.Config, buildInfo component.BuildInfo, logGroupName string, logRetention int64, sess *session.Session) *Client {
+func NewClient(logger *zap.Logger, awsConfig *aws.Config, buildInfo component.BuildInfo, logGroupName string, logRetention int64, tags map[string]*string, sess *session.Session) *Client {
 	client := cloudwatchlogs.New(sess, awsConfig)
 	client.Handlers.Build.PushBackNamed(handler.RequestStructuredLogHandler)
 	client.Handlers.Build.PushFrontNamed(newCollectorUserAgentHandler(buildInfo, logGroupName))
-	return newCloudWatchLogClient(client, logRetention, logger)
+	region := ""
+	if !reflect.ValueOf(awsConfig.Region).IsNil() {
+		region = *awsConfig.Region
+	}
+	stsClient := sts.New(sess, awsConfig)
```

Comment:
sts is tricky. Since we are using the aws sdk for go v1, we need to make sure that we use the regionalized endpoint for sts, otherwise we might have problems in some regions with high latency or failure.

Make sure that the config has this https://docs.aws.amazon.com/sdk-for-go/api/aws/#Config.WithSTSRegionalEndpoint

Reply:
Added STSRegionalEndpoint to the config in latest commit.